src/GafferImage/Constant.cpp

```diff
@@ -138,11 +138,15 @@ GafferImage::Format Constant::computeFormat( const Gaffer::Context *context, con
 void Constant::hashDataWindow( const GafferImage::ImagePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
 {
 	FlatImageSource::hashDataWindow( output, context, h );
-	h.append( formatPlug()->hash() );
+	h.append( formatPlug()->displayWindowPlug()->hash() );
```

Comment:
This is bypassing the work done by `FormatPlug::hash()` to fall back to a default format from the context if one isn't specified by the plug. It breaks the default format provided in the File->Settings window. Rather than try to get clever, let's revert this and the comment in `computeDataWindow()` and just switch `affects()` to use `formatPlug()->isAncestorOf()`. I don't think the negligible performance benefit of omitting `pixelAspectPlug()` from the hash justifies anything more complex.